src/main/java/io/github/cottonmc/cotton/gui/widget/WToggleButton.java

```diff
@@ -11,57 +11,122 @@
 import net.minecraft.util.Identifier;
 
 public class WToggleButton extends WWidget {
-	Text label = null;
-	Identifier off_image = new Identifier("libgui:widget/toggle_off.png");
-	Identifier on_image  = new Identifier("libgui:widget/toggle_on.png");
-	boolean on = false;
-	protected int color = WLabel.DEFAULT_TEXT_COLOR;
-	protected int darkmodeColor = WLabel.DEFAULT_DARKMODE_TEXT_COLOR;
-	
+
+
+	/** All default values, no text */
 	public WToggleButton() {
-		
+		this(DEFAULT_ON_IMAGE, DEFAULT_OFF_IMAGE, 18, 8);
 	}
-	
+
+	/** Defaults with text */
 	public WToggleButton(Text text) {
+		this(DEFAULT_ON_IMAGE, DEFAULT_OFF_IMAGE, 18, 8);
 		this.label = text;
 	}
-	
+
+	/** Custom images,  with default sizes */
+	public WToggleButton(Identifier onImage, Identifier offImage) {
+
+		this.onImage = onImage;
+		this.offImage = offImage;
+	}
+
+	/** Custom images,  with default sizes and a label */
+	public WToggleButton(Identifier onImage, Identifier offImage, Text label) {
+
+		this.onImage = onImage;
+		this.offImage = offImage;
+		this.label = label;
+	}
+
+	/** Custom images,  with custom sizes */
+	public WToggleButton(Identifier onImage, Identifier offImage, int width, int height) {
+
+		this.onImage = onImage;
+		this.offImage = offImage;
+		this.width = width;
+		this.height = height;
+	}
+
+	/** Fully customized,  custom images,  sizes and label */
+	public WToggleButton(Text label, Identifier onImage, Identifier offImage, int width, int height) {
+		this(onImage, offImage, width, height);
+
+		this.label = label;
+	}
+
 	public WToggleButton color(int light, int dark) {
+
 		this.color = light;
 		this.darkmodeColor = dark;
+
 		return this;
 	}
-	
+
+	Text label = null;
+
+	/** Default On / Off Images */
+	protected final static Identifier DEFAULT_OFF_IMAGE = new Identifier("libgui:widget/toggle_off.png");
+	protected final static Identifier DEFAULT_ON_IMAGE  = new Identifier("libgui:widget/toggle_on.png");
+
+	protected Identifier onImage;
```

Comment:
thanks for making these fields protected. We should also make isOn and onToggle protected just for subclass support (or private if getter access is preferred)